File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Get input from user
    string text = get_string("Text: ");

    // Get counters accordingly
    int lettersCounter = count_letters(text);
    int wordsCounter = count_words(text);
    int sentencesCounter = count_sentences(text);

    // Initialize vars
    float averageLetters = 0;
    float averageSentences = 0;
    int liauIndex = 0;

    // Calculate average letters and sentences by converting letters to a float, in the end
    // divide letters and words and multiply them by 100
    averageLetters = (float)lettersCounter / wordsCounter * 100;
    averageSentences = (float)sentencesCounter / wordsCounter * 100;

    // Calculate liauIndex, make sure to round the result to the nearest integer
    liauIndex = round(0.0588 * averageLetters - 0.296 * averageSentences - 15.8);

    // Output result accordingly
    if (liauIndex < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (liauIndex > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", liauIndex);
    }
}

// Get total number of letters using isalpha method
int count_letters(string text)
{
    int counter = 0;

    for (int i = 0; i < strlen(text); i++)
    {

        if (isalpha(text[i]))
        {
            counter++;
        }
    }

    return counter;
}

// Get total number of word using isspace method
int count_words(string text)
{
    int counter = 1;

    for (int i = 0; i < strlen(text); i++)
    {

        if (isspace(text[i]))
        {
            counter++;
        }
    }

    return counter;
}

// Get total number of sentences by checking if a letter is a dot, esclamation mark, or question mark
int count_sentences(string text)
{
    int counter = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        int textToInt = (int)text[i];

        if (textToInt == 46 || textToInt == 33 || textToInt == 63)
        {
            counter++;
        }
    }

    return counter;
}

File: sentimental-readability/readability.py
def main():
    text = input("Text: ")

    letters_counter = count_letters(text)
    words_counter = count_words(text)
    sentences_counter = count_sentences(text)

    # Calculate average letters and sentences by diving both with words_counter and multiplying by 100
    average_letters = letters_counter / words_counter * 100
    average_sentences = sentences_counter / words_counter * 100

    # Calculate liauIndex, while making sure to round the result to the nearest integer
    liau_index = round(0.0588 * average_letters - 0.296 * average_sentences - 15.8)

    # Output result accordingly
    if liau_index < 1:
        print("Before Grade 1")

    elif liau_index > 16:
        print("Grade 16+")

    else:
        print(f"Grade {liau_index}")


# Get total number of letters using isalpha method
def count_letters(text):
    counter = 0

    for word in text:

        if word.isalpha():
            counter += 1

    return counter


# Get total number of words using isspace method
def count_words(text):
    counter = 1

    for word in text:

        if word.isspace():
            counter += 1

    return counter


# Get total number of sentences by checking if a letter is a dot, esclamation mark, or question mark
def count_sentences(text):
    counter = 0

    for word in text:

        # Source code for getting ASCII value of a word: https://www.delftstack.com/howto/python/convert-string-to-ascii-python/
        word_to_int = ord(word)

        if word_to_int == 46 or word_to_int == 33 or word_to_int == 63:
            counter += 1

    return counter


main()
